app/pages/index.tsx

```diff
@@ -27,41 +28,87 @@ const Home: NextPage = () => {
   const connectedWallet = useAnchorWallet();
 
   const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [creator, setCreator] = useState<ProgramAccount | null>(null);
+  const { user, setUser } = useUser();
 
   const program = useMemo(() => {
     if (connectedWallet) {
       const provider = new anchor.Provider(connection, connectedWallet, OPTS);
 
-      return new anchor.Program(IDL as anchor.Idl, PROGRAM_ID, provider);
+      return new anchor.Program<NftClub>(
+        IDL as unknown as NftClub,
+        PROGRAM_ID,
+        provider
+      );
     }
 
     return null;
   }, [connectedWallet]);
 
-  const getCreatorAccountForUserWallet = async (
-    nftClubProgram: anchor.Program,
-    wallet: AnchorWallet
-  ) => {
-    const [creator] = await nftClubProgram.account.creator.all([
-      {
-        memcmp: {
-          offset: 8, // Discriminator.
-          bytes: wallet.publicKey.toBase58(),
+  const getCreatorAccountForUserWallet = useCallback(
+    async (
+      nftClubProgram: anchor.Program<NftClub>,
+      wallet: AnchorWallet
+    ): Promise<Record<string, unknown>> => {
+      const creatorSeeds = [
+        wallet.publicKey.toBuffer(),
+        anchor.utils.bytes.utf8.encode('creator'),
+      ];
+
+      const [creatorPubKey] = await anchor.web3.PublicKey.findProgramAddress(
+        creatorSeeds,
+        nftClubProgram.programId
+      );
+
+      return await nftClubProgram.account.creator.fetch(creatorPubKey);
+    },
+    []
+  );
+
+  const fetchSubscriptionsForUserWallet = useCallback(
+    async (
+      nftClubProgram: anchor.Program<NftClub>,
+      wallet: AnchorWallet
+    ): Promise<Record<string, unknown>[]> => {
+      return await nftClubProgram.account.subscription.all([
+        {
+          memcmp: {
+            offset: 8,
+            bytes: wallet.publicKey.toBase58(),
+          },
         },
-      },
-    ]);
+      ]);
+    },
+    []
+  );
+
+  const fetchUserDetails = useCallback(
+    async (nftClubProgram: anchor.Program<NftClub>, wallet: AnchorWallet) => {
+      const creator = await getCreatorAccountForUserWallet(
+        nftClubProgram,
+        wallet
+      );
+      const subscriptions = await fetchSubscriptionsForUserWallet(
+        nftClubProgram,
+        wallet
+      );
 
-    if (creator) setCreator(creator);
-    setIsLoading(false);
-  };
+      (creator || subscriptions.length) &&
+        setUser({
+          subscriptions,
+          creatorAccount: creator,
+        });
```

Comment:
It may be worth moving all of this logic up a level to `_app.tsx` because no matter what route a user first hits, we'll want to fetch this kind of data